### PR TITLE
added match_atoms keyword to analysis.align

### LINKED ISSUE #2285: option to turn off the mass match in alignto()
**Is your feature request related to a problem? Please describe.**
When one might want to convert the coarse-grain simulation to atomistic simulation, one will usually align the crystallographic protein to the coarse-grain protein. 

**Describe the solution you'd like**
Ideally, this should be done using 

```
from MDAnalysis.analysis.align import alignto
alignto(AT.atoms, CG.atoms, select=('name CA', 'name BB'))
```

However, since the mass of the CG backbone bead has a different mass from the CA atoms, the error `MDAnalysis.exceptions.SelectionError: Inconsistent selections, masses differ by more than 0.1; mis-matching atoms are shown above.`
 is raised and there isn't a way of turning this off.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #2285 

Changes made in this Pull Request:
 - added keyword `match_atoms` to turn off n_residue and mass matching

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/align.py [hunk 1]
```diff
@@ -453,6 +456,8 @@ def alignto(mobile, reference, select="all", weights=None,
     .. _ClustalW: http://www.clustal.org/
     .. _STAMP: http://www.compbio.dundee.ac.uk/manuals/stamp.4.2/
 
+    .. versionchanged:: 0.20.1
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: Please change to 0.21.0

File: package/MDAnalysis/analysis/align.py [hunk 2]
```diff
@@ -471,7 +476,7 @@ def alignto(mobile, reference, select="all", weights=None,
     .. versionchanged:: 0.17.0
        Deprecated keyword `mass_weighted` was removed.
     """
-    if select in ('all', None):
+    if select in ('all', None): # legacy
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: Why "legacy" comment?

File: package/MDAnalysis/analysis/align.py [hunk 3]
```diff
@@ -1040,16 +1054,21 @@ def get_matching_atoms(ag1, ag2, tol_mass=0.1, strict=False):
     """
 
     if ag1.n_atoms != ag2.n_atoms:
+        if not match_atoms:
+            raise SelectionError("Mobile and reference atom selections do not "
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/align.py]: We log  the error below via `logger.error` so this one should also be logged (for consistency).

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 4]
```diff
@@ -127,6 +127,16 @@ def test_nomatch_residues_raise_empty(self, universe, reference_small,
             with pytest.warns(SelectionWarning):
                 with pytest.raises(SelectionError):
                     groups = align.get_matching_atoms(ref, mobile, strict=strict)
+    
+    def test_toggle_atom_match(self, universe, reference):
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/test_align.py]: This should be 3 different tests.

File: package/MDAnalysis/analysis/align.py (hunk 2 above)
Comment (lilyminium) [reply package/MDAnalysis/analysis/align.py]: Ugh, leftover from code I rewrote. Removed now.

**Comment by lilyminium** [comment]
Thanks @orbeckst, does this work?

**Comment by orbeckst** [comment]
Let's see what the coverage says.

**Comment by orbeckst** [comment]
I merged (and removed an old comment). Ready to squash-merge once CI is green again.

Apologies for the delay in reviewing.

**Comment by lilyminium** [comment]
Thanks!

**Comment by orbeckst** [comment]
I merged develop and didn't wait for CI as this was only a conflict in CHANGELOG. Thanks for the fix!

**Pull request merged**